keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt

```diff
@@ -16,8 +16,13 @@
 package com.netflix.spinnaker.keel.events
 
 import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.SubmittedResource
+import de.huxhorn.sulky.ulid.ULID
 
-data class ResourceEvent(
-  val type: ResourceEventType,
-  val resource: Resource<*>
-)
+sealed class ResourceEvent
```

Comment:
Replaced a single event class with a `type` field with 3 different event sub-types as I want each one to have a different payload.